validation-test/stdlib/Lazy.swift.gyb

```diff
@@ -1279,5 +1279,158 @@ do {
+  ([0, 10, 20, 30, 40],     99,     5) ]
+
+tests.test("LazyPrefixWhileIterator") {
+  let base = (1...10).lazy.prefix(while: { $0 < 4 })
+
+  var iter1 = base.makeIterator()
+  expectEqual(1, iter1.next())
+  expectEqual(2, iter1.next())
+  expectEqual(3, iter1.next())
+  expectEqual(nil, iter1.next())
+}
+
+% for Kind in 'Sequence', 'Forward', 'Bidirectional':
+%   Self = 'Sequence' if Kind == 'Sequence' else collectionForTraversal(Kind)
+%   checkKind = 'ForwardCollection' if Kind == 'Forward' else Self
+tests.test("LazyPrefixWhile${Self}").forEach(in: prefixDropWhileTests) {
+  (data, value, pivot) in
+
+  let base = Minimal${Self}(elements: data)
+
+  var calls1 = 0
+  let prefixed = base.lazy.prefix(while: { calls1 += 1; return $0 != value })
+  let expected = data.prefix(upTo: pivot)
+  expectEqual(0, calls1)
+  check${checkKind}(expected, prefixed)
+
+  var calls2 = 0
+  _ = prefixed.map { _ in calls2 += 1 }
+  expectEqual(0, calls2, "unexpected eagerness in \(type(of: prefixed))")
+
+  % if Kind == 'Bidirectional':
+  check${checkKind}(expected.reversed(), prefixed.reversed())
+  % end
+}
+% end
+
+tests.test("LazyPrefixWhileSequence/AssociatedTypes") {
+  typealias Base = MinimalSequence<OpaqueValue<Int>>
+  typealias Subject = LazyPrefixWhileSequence<Base>
+  expectSequenceAssociatedTypes(
+    sequenceType: Subject.self,
+    iteratorType: LazyPrefixWhileIterator<Base.Iterator>.self,
+    subSequenceType: AnySequence<OpaqueValue<Int>>.self)
+}
+
+tests.test("LazyPrefixWhileCollection/AssociatedTypes") {
+  typealias Base = MinimalCollection<OpaqueValue<Int>>
+  typealias Subject = LazyPrefixWhileCollection<Base>
+  expectCollectionAssociatedTypes(
+    collectionType: Subject.self,
+    iteratorType: LazyPrefixWhileIterator<Base.Iterator>.self,
+    subSequenceType: Slice<Subject>.self,
+    indexType: LazyPrefixWhileIndex<Base>.self,
+    indexDistanceType: Base.IndexDistance.self,
+    indicesType: DefaultIndices<Subject>.self)
+}
+
+tests.test("LazyPrefixWhileBidirectionalCollection/AssociatedTypes") {
+  typealias Base = MinimalBidirectionalCollection<OpaqueValue<Int>>
+  typealias Subject = LazyPrefixWhileBidirectionalCollection<Base>
+  expectBidirectionalCollectionAssociatedTypes(
+    collectionType: Subject.self,
+    iteratorType: LazyPrefixWhileIterator<Base.Iterator>.self,
+    // FIXME(ABI): SubSequence should be `LazyFilterBidirectionalCollection<Base.Slice>`.
+    subSequenceType: BidirectionalSlice<Subject>.self,
+    indexType: LazyPrefixWhileIndex<Base>.self,
+    indexDistanceType: Base.IndexDistance.self,
+    indicesType: DefaultBidirectionalIndices<Subject>.self)
+}
+
+//===--- LazyDropWhile ----------------------------------------------------===//
+
+tests.test("LazyDropWhileIterator") {
+  let base = (1...10).lazy.drop(while: { $0 < 4 })
+
+  var iter1 = base.makeIterator()
+  expectEqual(4, iter1.next())
+  expectEqual(5, iter1.next())
+  expectEqual(6, iter1.next())
+  expectEqual(7, iter1.next())
+}
```

Comment:
Same here, as long as you call `checkSequence()` below, you don't need this test.